### PR TITLE
fix(testbench): improve Python 3 compatibility

### PR DESCRIPTION
**Pull request by crwilcox**
Addresses some incompatibilities with Python 3. With these changes I was able to use Python 3 to upload a file to the test bench.

Related bug: #3056

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/3204)

### CODE REVIEW

File: ci/kokoro/docker/Dockerfile.fedora [hunk 1]
```diff
@@ -36,15 +36,15 @@ RUN dnf makecache && dnf install -y \
     ncurses-term \
     openssl-devel \
     pkgconfig \
-    python \
+    python38 \
```
Comment (remyabel) [inline ci/kokoro/docker/Dockerfile.fedora]: Why python38? Why not just install both python and python3? Most people will not have python38 installed. The current version of `python3` is `3.7.5`, which is also the default in Fedora 31.

File: ci/kokoro/docker/Dockerfile.fedora-install [hunk 2]
```diff
@@ -40,15 +40,15 @@ RUN dnf makecache && dnf install -y \
...
+    python38 \
     shtool \
     unzip \
     wget \
     which \
     zlib-devel
 
-RUN pip install --upgrade pip
-RUN pip install httpbin flask gevent gunicorn crc32c
+RUN python3.8 -m pip install --upgrade pip
```
Comment (remyabel) [inline ci/kokoro/docker/Dockerfile.fedora-install]: Genuine question: is it better to do `python3 -m pip` or `pip3` here?

File: ci/kokoro/docker/Dockerfile.fedora (hunk 1 above)
Comment (crwilcox) [reply ci/kokoro/docker/Dockerfile.fedora]: I am happy to rely on 3.7.5. I didn't count on there being one. happy to change this :)

File: ci/kokoro/docker/Dockerfile.fedora-install (hunk 2 above)
Comment (crwilcox) [reply ci/kokoro/docker/Dockerfile.fedora-install]: Not all OS platforms have `pip3`. Calling `python3 -m pip` ensures the pip we are using is the one connected to that particular interpreter. So the more consistent experience across platforms and double-sureness is why I tend to do this. In theory, they are equivalent.

**Comment by crwilcox** [comment]
@remyabel @coryan since opening this I figured out running the tests locally. If having this hang around is irritating, I can close it and reopen it once it is ready to merge. Just let me know.

**Comment by coryan** [comment]
Let's close it for now. Thanks for making these changes btw.

**Pull request closed without merging**